core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -99,6 +104,8 @@ public LoadBalancer newLoadBalancer(LoadBalancer.Helper helper) {
     @VisibleForTesting
     static final Attributes.Key<Ref<ConnectivityStateInfo>> STATE_INFO =
         Attributes.Key.create("state-info");
+    static final Attributes.Key<Ref<PickResult>> PICK_RESULT = Attributes.Key.create("pick-result");
```

Comment:
Why introduce `PICK_RESULT` attribute? I would rather not to store any state in Attributes unless I don't have any other solution. @zhangkun83 what do you think?

Reply:
Simply to avoid allocating an identical `PickResult` object every time the Subchannel is used. It's not "solving" anything, just reducing garbage production (_possibly_ by an insignificant amount in relative terms).